processing/src/main/java/org/apache/druid/segment/selector/settable/SettableValueDoubleColumnValueSelector.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment.selector.settable;
+
+import org.apache.druid.query.monomorphicprocessing.RuntimeShapeInspector;
+import org.apache.druid.segment.BaseDoubleColumnValueSelector;
+
+public class SettableValueDoubleColumnValueSelector implements BaseDoubleColumnValueSelector
```

Comment:
Hmm, is it possible to use existing `SettableDoubleColumnValueSelector`? Looks feasible if the value type conversion is done in some other selector. Not sure this way is better though.

Reply:
they are somewhat different , and that is harder to use in this context. that said, I have added comments on this class to explain its use.